workflow/controller/operator_test.go

```diff
@@ -3739,3 +3739,113 @@ func TestValidReferenceMode(t *testing.T) {
 	woc.operate()
 	assert.Equal(t, wfv1.NodeRunning, woc.wf.Status.Phase)
 }
+
+var workflowStatusMetric = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
```

Comment:
anyway to tidy this to just the relevant fields? hard to understand as it is

Reply:
This is as tidy as possible. This tests relies on using the raw K8s object as a snapshot of its state right before the behavior we are interested in testing.

Most of our operator tests use this pattern.